streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -115,13 +114,16 @@ public boolean isClosed() {
     @Override
     public void revive() {
         if (state == CLOSED) {
+            // clear all the stores since they should be re-registered
```

Comment:
Here's one of the tricky things that we need to pay attention to, since we need the task state-manager's changelog partitions for de-registration asynchronously at the restore-thread. And there are a few different scenarios:

1) completely closing the task: we remove the task from the task-manager, but materialized the state-manager's changelog partitions before clearing the stores since they need to be enqueued to the restore-thread.

2) reviving a corrupted task: we close the task but do NOT remove it from the task-manager, and we materialize the changelog partitions before clearing the stores.

3) recycling a task: we do NOT close the task but we do remove it from the task-manager, and hence we do not clear the state-manager's stores as well since we want to skip re-registering them. BUT we still need to materialize the changelog partitions to be enqueued to the restore-thread.